### PR TITLE
build: enable automatic test module teardown

### PR DESCRIPTION
**Pull request by crisbeto**
Angular 12.1.2 includes [opt-in automatic test module teardown](https://github.com/angular/angular/pull/42566) which is similar to our logic of destroying the current fixture. These changes clean up our monkey patches and use the new option instead.

### CODE REVIEW

File: src/components-examples/material/autocomplete/autocomplete-harness/autocomplete-harness-example.spec.ts [hunk 1]
```diff
@@ -14,7 +14,9 @@ describe('AutocompleteHarnessExample', () => {
   let loader: HarnessLoader;
 
   beforeAll(() => {
-    TestBed.initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting());
+    TestBed.initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting(), {
+      teardown: {destroyAfterEach: true}
```
Comment (josephperrott) [inline src/components-examples/material/autocomplete/autocomplete-harness/autocomplete-harness-example.spec.ts:18]: Can we also enable `rethrowErrors` everywhere as well?

```suggestion
      teardown: {destroyAfterEach: true, rethrowErrors: true}
```

Comment (crisbeto) [reply src/components-examples/material/autocomplete/autocomplete-harness/autocomplete-harness-example.spec.ts:18]: It default to rethrowing when `destroyAfterEach` is enabled: https://github.com/angular/angular/blob/master/packages/core/testing/src/r3_test_bed.ts#L482

**Review by devversion — APPROVED** [review]
👍 I assume a lint rule enforcing the option is not reasonable because eventually this will be the default anyway.

**Comment by crisbeto** [comment]
Yeah, the idea is to potentially flip the default in v13.

**Pull request merged**